{
  "url": "https://api.github.com/repos/shap/shap/issues/748",
  "repository_url": "https://api.github.com/repos/shap/shap",
  "labels_url": "https://api.github.com/repos/shap/shap/issues/748/labels{/name}",
  "comments_url": "https://api.github.com/repos/shap/shap/issues/748/comments",
  "events_url": "https://api.github.com/repos/shap/shap/issues/748/events",
  "html_url": "https://github.com/shap/shap/issues/748",
  "id": 480933843,
  "node_id": "MDU6SXNzdWU0ODA5MzM4NDM=",
  "number": 748,
  "title": "TypeError: shap_values() got multiple values for argument 'ranked_outputs'",
  "user": {
    "login": "SeungJinWang",
    "id": 46008351,
    "node_id": "MDQ6VXNlcjQ2MDA4MzUx",
    "avatar_url": "https://avatars.githubusercontent.com/u/46008351?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/SeungJinWang",
    "html_url": "https://github.com/SeungJinWang",
    "followers_url": "https://api.github.com/users/SeungJinWang/followers",
    "following_url": "https://api.github.com/users/SeungJinWang/following{/other_user}",
    "gists_url": "https://api.github.com/users/SeungJinWang/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/SeungJinWang/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/SeungJinWang/subscriptions",
    "organizations_url": "https://api.github.com/users/SeungJinWang/orgs",
    "repos_url": "https://api.github.com/users/SeungJinWang/repos",
    "events_url": "https://api.github.com/users/SeungJinWang/events{/privacy}",
    "received_events_url": "https://api.github.com/users/SeungJinWang/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 0,
  "created_at": "2019-08-14T23:39:23Z",
  "updated_at": "2019-08-14T23:39:23Z",
  "closed_at": null,
  "author_association": "NONE",
  "active_lock_reason": null,
  "body": "Dear Scott,\r\nI have three questions(Actually I have a lot of questions, but I will ask next time...)\r\n\r\nQ1. I tried to three inputs(categorical, continuous, text) and one output.\r\n\r\nfinal_model = Model([model1.input, model2.input, model3.input], model_merge_out)\r\nexplainer = shap.DeepExplainer(final_model, [X_categorical_npary[0:100], X_continuous_npary[0:100], X_padded_docs[0:100]])\r\nshap_values = explainer.shap_values([X_categorical_test_npary[0:100], X_continuous_test_npary[0:100], padded_docs_test[0:100]])\r\nshap.summary_plot(shap_values, [X_categorical_npary[0:100], X_continuous_npary[0:100], X_padded_docs[0:100]], plot_type=\"bar\")\r\n\r\nHowever, when I ran the last code  I got an error as below\r\n\r\nshap.summary_plot(shap_values, [X_categorical_npary[0:100], X_continuous_npary[0:100], X_padded_docs[0:100]], plot_type=\"bar\")\r\nTraceback (most recent call last):\r\n\r\n  File \"<ipython-input-26-0121926afd66>\", line 1, in <module>\r\n    shap.summary_plot(shap_values, [X_categorical_npary[0:100], X_continuous_npary[0:100], X_padded_docs[0:100]], plot_type=\"bar\")\r\n\r\n  File \"C:\\ProgramData\\Anaconda3\\lib\\site-packages\\shap\\plots\\summary.py\", line 75, in summary_plot\r\n    num_features = (shap_values[0].shape[1] if multi_class else shap_values.shape[1])\r\n\r\nAttributeError: 'list' object has no attribute 'shape'\r\n\r\nQ2. As above, whenever I run this kinds of multiple inputs model, I should get list type of \"shap_values\". So, I should convert list to numpy array type, right?\r\n\r\n\r\nQ3. When I tried to use two inputs as below\r\n\r\nfinal_model12 = Model([model1.input, model2.input], model_merge_out12)\r\nfinal_model12.compile(optimizer='adam', loss='mae', metrics=['mse', 'mae', 'mape'])\r\nhistory12 = final_model12.fit([X_categorical, X_continuous], Y_los, batch_size=64, epochs= 5 , verbose=1)\r\n\r\nexplainer_model12 = shap.DeepExplainer(final_model12, [X_categorical_npary[:100], X_continuous_npary[:100]])\r\nshap_values_model12 = explainer_model12.shap_values(X_categorical_test_npary[0:100], X_continuous_test_npary[:100], ranked_outputs=None, output_rank_order='max')\r\nshap.summary_plot(shap_values_model12, [X_categorical_npary[0:100], X_continuous_npary[0:100]], plot_type=\"bar\")\r\n\r\nAlso, I got an error as below\r\n\r\nTypeError: shap_values() got multiple values for argument 'ranked_outputs'\r\n\r\nThank you for your time.\r\n",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/shap/shap/issues/748/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/shap/shap/issues/748/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
